test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -120,7 +119,28 @@
                 {:keys [expected-fn query-fn]}
                 extraction-test-cases]
           (testing (format "extract %s function works as expected on %s column for driver %s" op col-type driver/*driver*)
-            (is (= (set (expected-fn op)) (set (test-temporal-extract (query-fn op field-id)))))))))))
+            (is (= (set (expected-fn op)) (set (test-temporal-extract (query-fn op field-id)))))))))
+
+    ;; not really sure why vertica is failing...
```

Comment:
What's the error with vertica?

Reply:
```
[Vertica][VJDBC](3459) ERROR: Function catalog.date_part(unknown, unknown) is not unique
```

even though the query has no `date_part` fn

```
SELECT "public"."times_mixed_times"."id" AS "id",
          "public"."times_mixed_times"."index" AS "index",
          "public"."times_mixed_times"."dt" AS "dt",
          "public"."times_mixed_times"."dt_tz" AS "dt_tz",
          "public"."times_mixed_times"."d" AS "d",
          CAST("public"."times_mixed_times"."as_dt" AS timestamp) AS "as_dt",
          CAST("public"."times_mixed_times"."as_d" AS date) AS "as_d",
          year(?) AS "get-year",
          CAST(extract(quarter
                       from ?) AS integer) AS "get-quarter",
          CAST(extract(month
                       from ?) AS integer) AS "get-month",
          CAST(extract(day
                       from ?) AS integer) AS "get-day",
          CASE
              WHEN ((dayofweek_iso(?) + 1) % 7) = 0 THEN 7
              ELSE ((dayofweek_iso(?) + 1) % 7)
          END AS "get-day-of-week",
          CAST(extract(hour
                       from ?) AS integer) AS "get-hour",
          CAST(extract(minute
                       from ?) AS integer) AS "get-minute",
          second(?) AS "get-second"
   FROM "public"."times_mixed_times"
```

Reply:
My guess is `extract` is syntactic sugar for `date_part`, and the error comes from not casting the timestamps correctly. Here's a minimal reproduction of the error:

```
(ffirst (:rows (dev/query-jdbc-db :vertica "select extract(epoch from '2022-10-01T01:30:00')")))
;; => Execution error (SyntaxErrorException) at com.vertica.util.ServerErrorData/buildException (REPL:-1).
;;    [Vertica][VJDBC](3459) ERROR: Function catalog.date_part(unknown, unknown) is not unique

(ffirst (:rows (dev/query-jdbc-db :vertica "select extract(epoch from '2022-10-01T01:30:00'::timestamp)")))
;; => 1664587800.000000M
```

Reply:
yep!.  fixed in https://github.com/metabase/metabase/pull/26706/commits/b2aeb6b237bdf12f4ca7d7c955d5be3af9b183d2